src/app/src/androidTest/java/com/example/bookmark/MyBooksActivityTest.java

```diff
@@ -35,7 +41,18 @@
     private Solo solo;
     @Rule
     public ActivityTestRule<MyBooksActivity> rule =
-        new ActivityTestRule<>(MyBooksActivity.class, true, true);
+        new ActivityTestRule<>(MyBooksActivity.class, true, false);
+
+    /**
+     * This class sets up the shared preferences prior to the activity being triggered.
+     */
+    @BeforeClass
+    public static void setUpSharedPref() {
+        SharedPreferences sharedPreferences = InstrumentationRegistry.getInstrumentation().getTargetContext()
+            .getSharedPreferences("LOGGED_IN_USER", MODE_PRIVATE);
```

Comment:
Could extract string "LOGGED_IN_USER" into a shared constant on the class that it is most related to, `MainActivity`.